appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/LicenseCheckerWebListener.java

```diff
@@ -0,0 +1,66 @@
+/*-
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full
+ * license.
+ */
+package com.vaadin.appsec.v7;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import javax.servlet.annotation.WebListener;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+import com.vaadin.pro.licensechecker.BuildType;
+import com.vaadin.pro.licensechecker.LicenseChecker;
+import com.vaadin.server.VaadinService;
+
+/**
+ * Session listener to verify the license.
+ */
+@WebListener
+public class LicenseCheckerWebListener implements HttpSessionListener {
+
+    static final String PROPERTIES_RESOURCE = "appsec-kit.properties";
+
+    static final String VERSION_PROPERTY = "appsec-kit.version";
+
+    static final String PRODUCT_NAME = "vaadin-appsec-kit";
+
+    @Override
+    public void sessionCreated(HttpSessionEvent se) {
+        VaadinService vaadinService = VaadinService.getCurrent();
+        if (vaadinService != null) {
```

Comment:
We probably need a check like https://github.com/vaadin/appsec-kit/pull/50/files#diff-03025132cdf82dcbb427eb830abcb7ddfaabd83f896d4f54a63bcc01a0322150R48-R60 here as well to prevent checking license multiple times unnecessarily.

Reply:
Now that is address in the same way, i.e. storing and checking service names to check the license once for each.